dfastmi/kernel/BedLevelCalculator.py

```diff
@@ -0,0 +1,144 @@
+from typing import List
+import numpy
+from dfastmi.kernel.typehints import Vector
+
+
+class BedLevelCalculator:
+    """
+    This class is used to calculate bed level related values.
+    """
+
+    def __init__(self, number_of_periods: int):
+        """
+        number_of_periods : int
+         Amount of the equilibrium bed level change for each respective discharge period available.
+        """
+        self.number_of_periods = number_of_periods
+
+    def get_element_wise_maximum(self, dzb: List[numpy.ndarray]) -> numpy.ndarray:
+        """
+        This method gets the element wise maximum from the given bed levels.
+
+        Arguments
+        ---------
+        dzb   : List[numpy.ndarray]
+            List of arrays containing the bed level change at the beginning of each respective discharge period.
+
+        Returns
+        -------
+        dzmax : numpy.ndarray
+            Maximum bed level change.
+        """
+        return numpy.maximum.reduce(dzb)
+
+    def get_element_wise_minimum(self, dzb: List[numpy.ndarray]) -> numpy.ndarray:
+        """
+        This method gets the element wise minimum from the given bed levels.
+
+        Arguments
+        ---------
+        dzb   : List[numpy.ndarray]
+            List of arrays containing the bed level change at the beginning of each respective discharge period.
+
+        Returns
+        -------
+        dzmin : numpy.ndarray
+            minimum bed level change.
+        """
+        return numpy.minimum.reduce(dzb)
+
+    def get_linear_average(self, number_of_days: Vector, dzb: List[numpy.ndarray]) -> numpy.ndarray:
```

Comment:
- This method can raise an Error if the length of `number_of_days` or `dzb` is less than `number_of_periods`.
- I know we talked about it shortly, but I am a little confused about the `number_of_days`. To me, it sounds as if it should be an integer (if I only look at the name), but it is a `Vector`.

Reply:
To answer the first part:
in the original main_computation you could find the following: 

```
    N = len(dzq)
    # N should also equal len(T) and len(rsigma)
```
From this I derived the constructor of this class and changed N into self.number_of_periods.
The "number_of_days" was previously "T", which is a vector and in turn a tuple[float,...].
The dzb is derived from N, and should be the same/similar size as dzq I think.

I wasnt sure up to what point I should have been putting validation on everything put into a method.
I can add the suggestion to the documentation.

Reply:
To answer the second part.
The "number_of_days" was previously "T", which is a vector and in turn a tuple[float,...].
I based the name on the documentation which was available regarding "T", it could be the name "number_of_days" isn't correct or is lacking.

```
    T : Vector
        A tuple of periods indicating the number of days during which each discharge applies.
```

I think one float in the tuple represents a 24h day, e.g. 0.5 would be 12h.
But I am not sure about this.
I made a comment in regard to the vector/tuple[float,...] in a review comment below.

Reply:
Discussed, first part will be discussed in dfast-101.
second part, ask Bert if the can explain more what T is, to get a better name for that.

Reply:
I will ask if Bert can review it based on variable naming

Reply:
I will update this.